src/controllers/reviewController.js

```diff
@@ -0,0 +1,88 @@
+const Review = require('../models/Review');
+const Album = require('../models/Album');
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+
+//create review
+const createReview = async (req, res) => {
+  //( frontend sends an album property in the req.body) ! we MUST PROVIDE which album we are requesting- "album": "64d2a94c793389a43fc5a8ec",
+  const { album: albumId } = req.body; //check /look for the album in req.body and assign it to albumId // in postman in REQUEST { album: albumId } : {"album": "647e247e69c32ece45e23978"}
```

Comment:
@akosasante I don't understand quier well this part- shouldn't it be just `router.route('/).post(authenticateUser, createReview); `instead of `router.route('/album/:albumId').post(authenticateUser, createReview);` ? because when we /api/v1/reviews/album/:albumId we don't actually **refer** to a specific album with this specific albumId ?

Reply:
I think it was fine to have `router.route('/album/:albumId').post(authenticateUser, createReview);` I was wondering in that case if we should be using that `:albumId` from the path in our controller.

So instead of looking in the body for `const { album: albumId } = req.body;`, we'd have `const { albumId } = req.params;`. Since the client is already telling us which album they want to create a review for based on the params and they don't need to repeat that info in the body. We would have to then make sure we do add it to `req.body` before creating the review; similar to what you're doing for the user id here: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/e76b227a58c8dc3ad5a0c65cd07c05c1a4385c83/src/controllers/reviewController.js#L27](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/e76b227a58c8dc3ad5a0c65cd07c05c1a4385c83/src/controllers/reviewController.js#L27)

The change you made to have the create endpoint be just on the `api/v1/reviews` route is fine as well though! In that case we have to get that album info from the body. :+1: